pkg/azureclients/armclient/azure_armclient.go

```diff
@@ -75,20 +76,35 @@ type Client struct {
 }
 
 // New creates a ARM client
-func New(authorizer autorest.Authorizer, baseURI, userAgent, apiVersion, clientRegion string, clientBackoff *retry.Backoff) *Client {
-	restClient := autorest.NewClientWithUserAgent(userAgent)
-	restClient.PollingDelay = 5 * time.Second
-	restClient.RetryAttempts = 3
-	restClient.RetryDuration = time.Second * 1
+func New(authorizer autorest.Authorizer, clientConfig azureclients.ClientConfig, baseURI, apiVersion string) *Client {
```

Comment:
using the client configs reduces the number of parameter but it feels odd to have it as a dependency from `azureclients`